### PR TITLE
test case for transform with fillna #27905

### LINKED ISSUE #27905: Functions allowed in DataFrameGroupby.transform are not all tested
e.g. `fillna`, `tshift`and `corrwith` are all allowed in `group.transform()`, but looking at `pandas/tests/groupby/test_transform.py`, they are not tested. There are more, and I will add to here and make a PR to tests these cases.

### PR DESCRIPTION
**Pull request by ukantesaria**
- [x] closes #27905 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_transform.py [hunk 1]
```diff
@@ -1074,3 +1074,32 @@ def test_transform_lambda_with_datetimetz():
         name="time",
     )
     assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "inputDF, expectedDF",
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_transform.py]: inputDF -> input_df
expectedDF -> expected_df

are there other dtypes we should be testing?

**Comment by charlesdong1991** [comment]
thanks for creating the PR, i think it would be nice also to test `groupby(group_col).transform("fillna")` case

**Comment by jreback** [comment]
can you merge master and update to comments

**Comment by jreback** [comment]
can you merge master and update

File: pandas/tests/groupby/test_transform.py [hunk 2]
```diff
@@ -1074,3 +1074,32 @@ def test_transform_lambda_with_datetimetz():
...
+                    "B": [1.0, 2.0, 2.0, 3.0, 3.0, 3.0, 4.0],
+                }
+            ),
+            1,
+        ),
+    ],
+)
+def test_groupby_transform_fillna(inputDF, expectedDF):
+    # GH 27905 - Test fillna in groupby.transform
+    input1 = inputDF.groupby("A").transform(lambda x: x.fillna(x.mean()))
```
Comment (jreback) [inline pandas/tests/groupby/test_transform.py]: use the forms

result =
tm.assert_frame_equal(result, expected)

**Review by jreback — CHANGES_REQUESTED** [review]
can you parameterize this over all of the functions in the OP

**Comment by jreback** [comment]
can you merge master and update to the comments

**Comment by WillAyd** [comment]
closing as stale

**Pull request closed without merging**